src/filters/FiltersActions.js

```diff
@@ -244,3 +245,78 @@ export const fetchFilterConfig = () => {
 
   };
 };
+
+const fetchDistributions = () => {
+  return {
+    type: FETCH_DISTRIBUTIONS
+  };
+};
+
+const fetchDistributionsSuccess = (data) => {
+  return {
+    type: FETCH_DISTRIBUTIONS_SUCCESS,
+    data: data
+  };
+};
+
+const fetchDistributionsError = (err) => {
+  return {
+    type: FETCH_DISTRIBUTIONS_ERROR,
+    data: err
+  };
+};
+
+
+const distributionForInput = (x, inputValue) => {
+  return x.addQuery()
+  // .match({['statistics.comments.all.all.'+inputValue]: {$lte: 15}})
+  .project({
+    count: {
+      $subtract: [
+        '$statistics.comments.all.all.'+inputValue,
+        {
+          $mod: ['$statistics.comments.all.all.'+inputValue, 15]
+        }
+      ]
+    },
+    _id: false
+  })
+  .group({
+    _id: '$count',
+    total: {$sum: 1}
+  })
+  .sort({
+    '_id': 1
+  });
+};
+
+const inputValues = [
+  'count',
+  'replied_count',
+  'replied_ratio',
+  'reply_count',
+  'reply_ratio',
+  'word_count_average'
+];
```

Comment:
I think instead of doing this query on the fly each time we can create the actual query and then just call it by name.